src/main/resources/templates/login/login.html

```diff
@@ -14,6 +14,9 @@
         <div class="col-md-4 col-md-offset-4">
             <div class="portlet portlet-green">
                 <div class="portlet-body">
+                    
+                    <div th:if="${param.error}" class="alert alert-danger">Invalid username or password</div>
```

Comment:
It is better to define `Invalid username or password` in the `i18n` files, so that we can rely on user language configuration to show messages.

Reply:
Okay I will do the changes.